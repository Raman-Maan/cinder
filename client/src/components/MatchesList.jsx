import React, { Component } from 'react';
import './styles/MatchesList.css';

import MatchesListItem from '../components/MatchesListItem';

class MatchesList extends Component {
  constructor(props) {
    super(props);

    this.state = {
      active: false
    };
  }

  onListItemClick(id) {
    this.setState({ active: id });
    this.props.clickHandler(id);
  }

  render() {
    if (!this.props.matches || !this.props.matches.length) {
      return (
        <div className="MatchesList">
          <div className="empty">
            <h3 className="msg align-self-center">
              No matches yet. Get swiping!
            </h3>
          </div>
        </div>
      );
    }

    const listItems = this.props.matches.map(
      ({ id, title, subtitle, img, date }) => (
        <MatchesListItem
          key={id}
          title={title}
          subtitle={subtitle}
          img={img}
          date={date}
          active={id === this.state.active}
          // eslint-disable-next-line react/jsx-no-bind
          onClick={this.onListItemClick.bind(this, id)}
        />
      )
    );

    return (
      <div className="MatchesList">
        <ul>{listItems}</ul>
      </div>
    );
  }
}

export default MatchesList;
