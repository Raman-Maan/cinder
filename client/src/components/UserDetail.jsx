import React from 'react';

import './styles/UserDetail.css';
import genderMale from '../assets/gender-male.svg';
import genderFemale from '../assets/gender-female.svg';
import genderOther from '../assets/gender-other.svg';

class UserDetail extends React.Component {
  getGenderPic() {
    if (!this.props.userDetail || !this.props.userDetail.userName) {
      return;
    }
    if (this.props.userDetail.userGender === 'Male') return genderMale;
    else if (this.props.userDetail.userGender === 'Female') return genderFemale;
    else return genderOther;
  }

  dateFormat() {
    return (
      this.props.userDetail.matchTime.toLocaleDateString() +
      ' ' +
      this.props.userDetail.matchTime.toLocaleTimeString()
    );
  }

  render() {
    if (!this.props.userDetail || !this.props.userDetail.userName) {
      return (
        <div className="UserDetail">
          <div className="empty">
            <h3 className="msg align-self-center">
              No match selected yet. Select one match from the list!
            </h3>
          </div>
        </div>
      );
    }

    return (
      <div className="UserDetail">
        <img
          className="userImage"
          src={this.props.userDetail.userPics}
          alt=""
        />
        <h2>{this.props.userDetail.userName}</h2>
        <div className="userAgeWithGender">
          <span className="userAge">{this.props.userDetail.userAge}</span>
          <img className="userGender" src={this.getGenderPic()} alt="" />
        </div>
        <div>
          <p>
            <span className="titleStyle">Matched:</span>{' '}
            <span>
              {this.props.userDetail.matchTime ? this.dateFormat() : 'Unkown'}{' '}
            </span>
          </p>
        </div>
        <div>
          <span className="titleStyle">Bio:</span>{' '}
          <p className="textAutoWrap">{this.props.userDetail.userBio}</p>
        </div>
      </div>
    );
  }
}

export default UserDetail;
