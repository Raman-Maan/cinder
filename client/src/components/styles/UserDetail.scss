$empty-msg-color: #aaa;

.UserDetail {
  height: 100%;
  width: 100%;

  text-align: center;

  .empty {
    display: inline-block;
    width: 100%;
    height: 100%;

    padding: 1em;

    justify-content: center;

    .msg {
      color: $empty-msg-color;
    }
  }

  .userImage {
    display: inline-block;
    max-width: 300px;
    max-height: 200px;
    width: auto;
    height: auto;
  }

  .userAgeWithGender {
    .userAge {
      display: inline-block;
      margin-right: 30px;
    }

    .userGender {
      display: inline-block;
      max-width: 30px;
      max-height: 20px;
      width: auto;
      height: auto;
    }
  }

  .titleStyle {
    font-weight: bold;
  }

  .textAutoWrap {
    margin: 0 auto;
    max-width: 500px;
    padding: 12px;
  }
}
