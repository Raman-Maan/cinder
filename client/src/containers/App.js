import React, { Component } from 'react';
import logo from '../images/logo.svg';
import './App.css';

class App extends Component {
  state = {
    server_response: ''
  };

  getMessage = () => {
    return fetch('/api/user/example')
      .then(res => res.json())
      .catch(err => console.error(err));
  };

  componentDidMount() {
    this.getMessage()
      .then(msg => this.setState({ server_response: msg.example }))
      .catch(err => console.error(err));
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">{this.state.server_response}</p>
      </div>
    );
  }
}

export default App;
