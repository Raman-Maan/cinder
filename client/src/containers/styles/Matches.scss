.Matches {
  display: flex;
  height: 100%;
  width: 100%;

  .container-fluid {
    .row.full {
      width: 100%;
      height: 100%;
      .col {
        padding: 0;
      }
    }
  }
}
