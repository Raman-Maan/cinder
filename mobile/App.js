import React from 'react';
import Expo, { AppLoading } from 'expo';

import Home from './containers/home';

export default class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      fontLoaded: false,
    };
  }

  componentDidMount() {
    this.loadAssets();
  }

  async loadAssets() {
    await Expo.Font.loadAsync({
      Roboto: require('native-base/Fonts/Roboto.ttf'),
      Roboto_medium: require('native-base/Fonts/Roboto_medium.ttf'),
    });

    this.setState({ fontLoaded: true });
  }

  render() {
    if (!this.state.fontLoaded) {
      return <AppLoading />;
    }

    return (
      <Home />
    );
  }
}
