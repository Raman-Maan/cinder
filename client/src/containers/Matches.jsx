import React, { Component } from 'react';
import { Container, Row, Col } from 'reactstrap';

import './styles/Matches.css';
import MatchesList from '../components/MatchesList';

class Matches extends Component {
  constructor(props) {
    super(props);

    this.fetchUserMatches = this.fetchUserMatches.bind(this);

    this.state = {
      matches: []
    };
  }

  componentDidMount() {
    this.fetchUserMatches();
  }

  fetchUserMatches() {
    this.setState({
      matches: [
        {
          title: 'Kendrick Lamar',
          subtitle: 'How u doin?',
          date: new Date().toJSON(),
          img: 'http://cache.umusic.com/_sites/kendricklamar.com/images/og.jpg'
        },
        {
          title: 'Mac Miller | Larry Fisherman | Delusional Thomas',
          subtitle:
            ';););) Want to meet hot new singles? Click this totally legit link: www.gethotsingles.com',
          date: new Date().toJSON(),
          img:
            'https://i.scdn.co/image/f4509fe9c589c12be5470653178f901bd697b97b'
        },
        {
          title: 'Ian Simpson',
          subtitle: 'What u wearin',
          date: new Date().toJSON(),
          img:
            'https://media.pitchfork.com/photos/592997c25e6ef9596931f65a/1:1/w_300/e2fc485c.jpg'
        }
      ]
    });
  }

  render() {
    return (
      <div className="Matches">
        <Container fluid>
          <Row className="full">
            <Col sm="4">
              <MatchesList matches={this.state.matches} />
            </Col>
            <Col sm="8">
              <h1>This is the Matches page</h1>
            </Col>
          </Row>
        </Container>
      </div>
    );
  }
}

export default Matches;
