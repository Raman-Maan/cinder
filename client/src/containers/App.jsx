import React, { Component } from 'react';
import logo from '../assets/logo.svg';
import './App.css';

class App extends Component {
  static getMessage() {
    return fetch('/api/user/example')
      .then(res => res.json())
      .catch(err => console.error(err));
  }

  constructor() {
    super();
    this.state = {
      server_response: 'No request for message has been made'
    };
  }

  componentDidMount() {
    App.getMessage()
      .then(msg => this.setState({ server_response: msg.example }))
      .catch(err => console.error(err));
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">{this.state.server_response}</p>
      </div>
    );
  }
}

export default App;
