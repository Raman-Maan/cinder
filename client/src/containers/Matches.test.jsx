import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import sinon from 'sinon';

import App from './App';
import Matches from './Matches';
import MatchesList from '../components/MatchesList';
import UserDetail from '../components/UserDetail';

describe('<Matches />', () => {
  it('renders without crashing', () => {
    // simple smoke test
    shallow(
      <MemoryRouter>
        <Matches />
      </MemoryRouter>
    );
  });

  it('renders correctly', () => {
    const component = renderer
      .create(
        <MemoryRouter>
          <Matches />
        </MemoryRouter>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('routes correctly', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/matches']}>
        <App />
      </MemoryRouter>
    );
    expect(wrapper.find(Matches)).toHaveLength(1);
  });

  it('should show the <MatchesList /> component be default', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/matches']}>
        <App />
      </MemoryRouter>
    );
    expect(wrapper.find(MatchesList)).toHaveLength(1);
  });

  it('should show the <UserDetail /> component be default', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={['/matches']}>
        <App />
      </MemoryRouter>
    );
    wrapper.setState({ userDetail: false });

    expect(wrapper.find(UserDetail)).toHaveLength(1);
  });
});
