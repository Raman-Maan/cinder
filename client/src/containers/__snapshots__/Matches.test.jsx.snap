// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Matches /> renders correctly 1`] = `
<div
  className="Matches"
>
  <div
    className="container-fluid"
  >
    <div
      className="full row"
    >
      <div
        className="col col-sm-4"
      >
        <div
          className="MatchesList"
        >
<<<<<<< HEAD
          <ul>
            <div
              className="MatchesListItem"
              onClick={[Function]}
              tabIndex="0"
            >
              <div
                className="img-container"
              >
                <img
                  alt=""
                  className="circular"
                  src="http://cache.umusic.com/_sites/kendricklamar.com/images/og.jpg"
                />
              </div>
              <div
                className="details"
              >
                <div
                  className="top"
                >
                  <div
                    className="left"
                  >
                    <span
                      className="title"
                    >
                      Kendrick Lamar
                    </span>
                  </div>
                  <div
                    className="right"
                  >
                    <span
                      className="date"
                    >
                      2018-02-12
                    </span>
                  </div>
                </div>
                <div
                  className="bottom"
                >
                  <span
                    className="subtitle"
                  >
                    How u doin?
                  </span>
                </div>
              </div>
            </div>
            <div
              className="MatchesListItem"
              onClick={[Function]}
              tabIndex="0"
            >
              <div
                className="img-container"
              >
                <img
                  alt=""
                  className="circular"
                  src="https://i.scdn.co/image/f4509fe9c589c12be5470653178f901bd697b97b"
                />
              </div>
              <div
                className="details"
              >
                <div
                  className="top"
                >
                  <div
                    className="left"
                  >
                    <span
                      className="title"
                    >
                      Mac Miller | Larry Fisherman | Delusional Thomas
                    </span>
                  </div>
                  <div
                    className="right"
                  >
                    <span
                      className="date"
                    >
                      2018-02-12
                    </span>
                  </div>
                </div>
                <div
                  className="bottom"
                >
                  <span
                    className="subtitle"
                  >
                    ;););) Want to meet hot new singles? Click this totally legit link: www.gethotsingles.com
                  </span>
                </div>
              </div>
            </div>
            <div
              className="MatchesListItem"
              onClick={[Function]}
              tabIndex="0"
            >
              <div
                className="img-container"
              >
                <img
                  alt=""
                  className="circular"
                  src="https://media.pitchfork.com/photos/592997c25e6ef9596931f65a/1:1/w_300/e2fc485c.jpg"
                />
              </div>
              <div
                className="details"
              >
                <div
                  className="top"
                >
                  <div
                    className="left"
                  >
                    <span
                      className="title"
                    >
                      Ian Simpson
                    </span>
                  </div>
                  <div
                    className="right"
                  >
                    <span
                      className="date"
                    >
                      2018-02-12
                    </span>
                  </div>
                </div>
                <div
                  className="bottom"
                >
                  <span
                    className="subtitle"
                  >
                    What u wearin
                  </span>
                </div>
              </div>
            </div>
            <div
              className="MatchesListItem"
              onClick={[Function]}
              tabIndex="0"
            >
              <div
                className="img-container"
              >
                <img
                  alt=""
                  className="circular"
                  src="http://cache.umusic.com/_sites/kendricklamar.com/images/og.jpg"
                />
              </div>
              <div
                className="details"
              >
                <div
                  className="top"
                >
                  <div
                    className="left"
                  >
                    <span
                      className="title"
                    >
                      Kendrick Lamar
                    </span>
                  </div>
                  <div
                    className="right"
                  >
                    <span
                      className="date"
                    >
                      2018-02-12
                    </span>
                  </div>
                </div>
                <div
                  className="bottom"
                >
                  <span
                    className="subtitle"
                  >
                    How u doin?
                  </span>
                </div>
              </div>
            </div>
            <div
              className="MatchesListItem"
              onClick={[Function]}
              tabIndex="0"
            >
              <div
                className="img-container"
              >
                <img
                  alt=""
                  className="circular"
                  src="https://i.scdn.co/image/f4509fe9c589c12be5470653178f901bd697b97b"
                />
              </div>
              <div
                className="details"
              >
                <div
                  className="top"
                >
                  <div
                    className="left"
                  >
                    <span
                      className="title"
                    >
                      Mac Miller | Larry Fisherman | Delusional Thomas
                    </span>
                  </div>
                  <div
                    className="right"
                  >
                    <span
                      className="date"
                    >
                      2018-02-12
                    </span>
                  </div>
                </div>
                <div
                  className="bottom"
                >
                  <span
                    className="subtitle"
                  >
                    ;););) Want to meet hot new singles? Click this totally legit link: www.gethotsingles.com
                  </span>
                </div>
              </div>
            </div>
            <div
              className="MatchesListItem"
              onClick={[Function]}
              tabIndex="0"
            >
              <div
                className="img-container"
              >
                <img
                  alt=""
                  className="circular"
                  src="https://media.pitchfork.com/photos/592997c25e6ef9596931f65a/1:1/w_300/e2fc485c.jpg"
                />
              </div>
              <div
                className="details"
              >
                <div
                  className="top"
                >
                  <div
                    className="left"
                  >
                    <span
                      className="title"
                    >
                      Ian Simpson
                    </span>
                  </div>
                  <div
                    className="right"
                  >
                    <span
                      className="date"
                    >
                      2018-02-12
                    </span>
                  </div>
                </div>
                <div
                  className="bottom"
                >
                  <span
                    className="subtitle"
                  >
                    What u wearin
                  </span>
                </div>
              </div>
            </div>
            <div
              className="MatchesListItem"
              onClick={[Function]}
              tabIndex="0"
            >
              <div
                className="img-container"
              >
                <img
                  alt=""
                  className="circular"
                  src="http://cache.umusic.com/_sites/kendricklamar.com/images/og.jpg"
                />
              </div>
              <div
                className="details"
              >
                <div
                  className="top"
                >
                  <div
                    className="left"
                  >
                    <span
                      className="title"
                    >
                      Kendrick Lamar
                    </span>
                  </div>
                  <div
                    className="right"
                  >
                    <span
                      className="date"
                    >
                      2018-02-12
                    </span>
                  </div>
                </div>
                <div
                  className="bottom"
                >
                  <span
                    className="subtitle"
                  >
                    How u doin?
                  </span>
                </div>
              </div>
            </div>
            <div
              className="MatchesListItem"
              onClick={[Function]}
              tabIndex="0"
            >
              <div
                className="img-container"
              >
                <img
                  alt=""
                  className="circular"
                  src="https://i.scdn.co/image/f4509fe9c589c12be5470653178f901bd697b97b"
                />
              </div>
              <div
                className="details"
              >
                <div
                  className="top"
                >
                  <div
                    className="left"
                  >
                    <span
                      className="title"
                    >
                      Mac Miller | Larry Fisherman | Delusional Thomas
                    </span>
                  </div>
                  <div
                    className="right"
                  >
                    <span
                      className="date"
                    >
                      2018-02-12
                    </span>
                  </div>
                </div>
                <div
                  className="bottom"
                >
                  <span
                    className="subtitle"
                  >
                    ;););) Want to meet hot new singles? Click this totally legit link: www.gethotsingles.com
                  </span>
                </div>
              </div>
            </div>
          </ul>
        </div>
      </div>
      <div
        className="col col-sm-8"
      >
        <div
          className="UserDetail"
        >
=======
>>>>>>> master
          <div
            className="empty"
          >
            <h3
              className="msg align-self-center"
            >
              No matches yet. Get swiping!
            </h3>
          </div>
        </div>
      </div>
      <div
        className="col col-sm-8"
      >
        <div>
          <h1>
            There's no one new around you.
          </h1>
        </div>
      </div>
    </div>
  </div>
</div>
`;
